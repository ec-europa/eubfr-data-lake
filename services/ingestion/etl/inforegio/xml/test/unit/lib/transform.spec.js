/**
 * @jest-environment node
 */

import mapper from '../../../src/lib/transform';
import testRecord from '../../stubs/record.json';
import testRecordBudget from '../../stubs/recordBudget.json';

describe('DG INFOREGIO XLS transformer', () => {
  let result = {};
  let resultBudget = {};

  beforeAll(() => {
    result = mapper(testRecord);
    resultBudget = mapper(testRecordBudget);
  });

  test('Returns null when record is not provided', () => {
    expect(mapper()).toBe(null);
  });

  test('Produces correct JSON output structure', () => {
    expect(result).toMatchSnapshot();
  });

  test('Budget values do not contain decimals', () => {
    expect(result.budget.eu_contrib.value % 1).toBe(0);
    expect(result.budget.total_cost.value % 1).toBe(0);
  });

  test('Preserves badly formatted budget', () => {
    expect(resultBudget).toMatchSnapshot();
  });
});
