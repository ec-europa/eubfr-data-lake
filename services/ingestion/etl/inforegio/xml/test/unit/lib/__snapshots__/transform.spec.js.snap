// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DG INFOREGIO XLS transformer Preserves badly formatted budget 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "",
      "raw": "20 EUR 40",
      "value": 0,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "",
      "raw": "FEDER (Objectif 5b 1994-1996) 60 979 euros (400 000 FRF)",
      "value": 0,
    },
  },
  "call_year": "",
  "comments": "",
  "complete": true,
  "description": "The colour of the sky never had much effect on the irksome rail trip between Greece's two principal cities of Athens and Thessaloniki: passengers had to wait for hours since there was only one set of tracks for trains travelling in both directions. Now all this has changed thanks to a huge programme carried out with the help of the European Union.",
  "ec_priorities": Array [],
  "media": Array [],
  "period": "2000-2006",
  "programme_name": "",
  "project_id": "75",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "GR",
      "location": null,
      "nuts": Array [
        Object {
          "code": "EL30",
          "level": null,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "",
      "region": "GREECE",
      "town": "",
    },
  ],
  "project_website": "http://ec.europa.eu/regional_policy/en/projects/greece/on-the-tracks-of-hercules",
  "related_links": Array [],
  "reporting_organisation": "REGIO",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [
    "Transport",
  ],
  "third_parties": Array [],
  "timeframe": Object {
    "from": null,
    "from_precision": "day",
    "to": null,
    "to_precision": "day",
  },
  "title": "On the tracks of Hercules",
  "type": Array [
    "Standard",
  ],
}
`;

exports[`DG INFOREGIO XLS transformer Produces correct JSON output structure 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "EUR",
      "raw": "EUR 157 000 000",
      "value": 157000000,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "",
      "raw": "FEDER (Objectif 5b 1994-1996) 60 979 euros (400 000 FRF)",
      "value": 0,
    },
  },
  "call_year": "",
  "comments": "",
  "complete": true,
  "description": "The colour of the sky never had much effect on the irksome rail trip between Greece's two principal cities of Athens and Thessaloniki: passengers had to wait for hours since there was only one set of tracks for trains travelling in both directions. Now all this has changed thanks to a huge programme carried out with the help of the European Union.",
  "ec_priorities": Array [],
  "media": Array [],
  "period": "2000-2006",
  "programme_name": "",
  "project_id": "75",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "GR",
      "location": null,
      "nuts": Array [
        Object {
          "code": "EL30",
          "level": null,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "",
      "region": "GREECE",
      "town": "",
    },
  ],
  "project_website": "http://ec.europa.eu/regional_policy/en/projects/greece/on-the-tracks-of-hercules",
  "related_links": Array [],
  "reporting_organisation": "REGIO",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [
    "Transport",
  ],
  "third_parties": Array [],
  "timeframe": Object {
    "from": null,
    "from_precision": "day",
    "to": null,
    "to_precision": "day",
  },
  "title": "On the tracks of Hercules",
  "type": Array [
    "Standard",
  ],
}
`;
